lbz/authz.py

```diff
@@ -166,31 +166,28 @@ def sign_authz(authz_data: dict, private_key_jwk: dict) -> str:
         )
 
 
-def check_permission(resource: Resource, permission_name: str) -> dict:
+def check_permission(resource: Union[Type[Resource], Resource], permission_name: str) -> dict:
     """
     Check if requester has sufficient permissions to do something on specific resource.
 
     Raises if not.
     """
-    authorization_header = resource.request.headers.get("Authorization")
-    authorization_scope = None
-    if not authorization_header:
-        if hasattr(resource, "get_guest_authorization"):
-            authorization_scope = resource.get_guest_authorization()
-        else:
-            raise Unauthorized("Authorization header missing or empty")
+    authorization_header = resource.request.headers.get("Authorization", "")
+    authorization_scope = resource.get_guest_authorization()
+    if not authorization_header and not authorization_scope:
+        raise Unauthorized("Authorization header missing or empty")
 
     authorizer = Authorizer(
         auth_jwt=authorization_header,
-        resource_name=getattr(resource, "_name") or resource.__class__.__name__.lower(),
+        resource_name=resource.get_name(),
         permission_name=permission_name,
         policy_override=authorization_scope,
     )
     authorizer.check_access()
     return authorizer.restrictions
 
 
-def has_permission(resource: Resource, permission_name: str) -> bool:
+def has_permission(resource: Union[Type[Resource], Resource], permission_name: str) -> bool:
```

Comment:
Is there really a case such that the class object is passed instead of an instance?

Reply:
If so, we have a bigger problem because one of the first lines tries to use request which is an instance attribute 😬 
```python
def check_permission(resource: Union[Type[Resource], Resource], permission_name: str) -> dict:
    authorization_header = resource.request.headers.get("Authorization", "")
    ...


def has_permission(resource: Union[Type[Resource], Resource], permission_name: str) -> bool:
    try:
        check_permission(resource, permission_name)
        ...
```